server/src/lib.rs

```diff
@@ -1,54 +1,59 @@
 use schedulers::stellarobject_timers::{self};
-use spacetimedb::{ReducerContext, Table};
-use types::common::{global_config, GlobalConfig};
+use spacetimedb::ReducerContext;
+use types::common::{global_config, CreateGlobalConfigRow, GetAllGlobalConfigRows, GetCountOfGlobalConfigRows, GlobalConfig, UpdateGlobalConfigRowById};
+use spacetimedsl::dsl;
 
 pub mod types;
 pub mod schedulers;
 
 #[spacetimedb::reducer(init)]
-pub fn init(ctx: &ReducerContext) {
+pub fn init(ctx: &ReducerContext) -> Result<(), String> {
+    let dsl = dsl(ctx);
+
     log::info!("init identity: {:?}", ctx.identity());
     log::info!("init sender: {:?}", ctx.sender);
 
     stellarobject_timers::init(ctx);
 
     // Create a Global Config row, or reinitalize the one if it exists.
-    if ctx.db.global_config().iter().count() == 0 {
-        ctx.db.global_config().insert(GlobalConfig {id:0, active_players:0, old_gods_defeated: 0 });
+    if dsl.get_count_of_global_configurations() == 0 {
+        dsl.create_global_config(0, 0, 0)?;
     } else {
-        if let Some(config) = ctx.db.global_config().id().find(0) {
-            ctx.db.global_config().id().update(GlobalConfig {
-                active_players: 0,
-                ..config
-            });
-        }
+        let mut config = dsl.get_all_global_configurations().last().ok_or("Failed to find existing global configuration")?;
+        config.set_active_players(0);
+        dsl.update_global_config_by_id(config)?;
     }
+    Ok(())
 }
 
 #[spacetimedb::reducer(client_connected)]
-pub fn identity_connected(ctx: &ReducerContext) {
+pub fn identity_connected(ctx: &ReducerContext) -> Result<(), String> {
+    let dsl = dsl(ctx);
     // Called everytime a new client connects
     
     // TODO: When someone logs in set their player to online
 
-    if let Some(config) = ctx.db.global_config().id().find(0) {
-        ctx.db.global_config().id().update(GlobalConfig {
-            active_players: config.active_players+1,
-            ..config
-        });
+    if let Some(config) = dsl.get_all_global_configurations().last().as_mut() {
```

Comment:
`if let mut Some(config) = dsl.get_all_global_configurations().last()`
(added `mut` after `let`, removed `.as_mut()`)

Reply:
Ah yeah, one week without rust and you forget the syntax order. 🤣 Glad you could remove the cloning though!